### ISSUE TITLE
A fleet of parked red cars magically appears

### ISSUE THREAD

**Issue by Elesbaan70** [CitiesSkylinesMods/TMPE#1603]
### Describe the problem

A fleet of parked red cars appears when roads are drawn in specific locations on specific maps. The problem occurs only when TMPE is enabled, and has been reproduced with no other mods enabled except Harmony and LOM.

### Steps to reproduce

1. Make sure TMPE is enabled.
2. Create a New Game based on the Black Woods map.
3. Zoom out and set your viewing angle so you have an overview of the initial tile.

![image](https://user-images.githubusercontent.com/95266227/174496187-40605505-dfb7-45e4-909e-c52e47c5cadd.png)

4. Find this rock in the middle of the tile.

![image](https://user-images.githubusercontent.com/95266227/174496305-5a3609e0-70dd-4b49-bfc0-f331d364d158.png)

5. Zoom in on its surrounding area.

![image](https://user-images.githubusercontent.com/95266227/174496346-ab7934d0-af89-43ae-98d9-e31b9e400f23.png)

6. Place a new two-lane road in this exact position relative to the rock and surrounding trees.

![image](https://user-images.githubusercontent.com/95266227/174496389-34d7d7f7-789f-4301-aba3-1204ac42cfeb.png)

7. Voilà!

![image](https://user-images.githubusercontent.com/95266227/174496403-64b68294-7889-4266-88c2-920b25655f66.png)

### Log files

[TMPE.log](https://ufile.io/fdvo8va0)
[output_log.txt](https://ufile.io/1v3ampme)

### Savegame?

The vanilla Black Woods map.

### Screenshots?

From [Reddit](https://www.reddit.com/r/CitiesSkylines/comments/vf930o/i_dont_know_why_but_all_cars_in_one_of_the/?utm_source=share&utm_medium=web2x&context=3):

![image](https://user-images.githubusercontent.com/95266227/174495707-110cc8af-9dd9-4f51-94d6-ae777394709f.png)

From @chimerabyte:

![image](https://user-images.githubusercontent.com/95266227/174495751-0e9eedcd-de37-493e-a1f0-f57c2bcb5e98.png)

From @kianzarrin:

![image](https://user-images.githubusercontent.com/26344691/105286348-c01d6d80-5bbe-11eb-873f-4825aaa0ccb7.png)

### Notes or questions?

Speculation on Reddit that this would only happen with enhanced parking enabled was incorrect. It happens any time TMPE is enabled.

**Comment by Elesbaan70** [comment]
Based on Reddit comments [here](https://www.reddit.com/r/CitiesSkylines/comments/vf930o/comment/icv2dqc/?utm_source=share&utm_medium=web2x&context=3), I think this is related to code that tries to make sure vehicles are actually at a valid location.

**Comment by krzychu124** [comment]
It's duplicate of #598 bug in the map settings. Already reported on the paradox forum.

> I think this is related to code that tries to make sure vehicles are actually at a valid location.

Parking location is implicitly valid once set. There is no code for later validation because it's expensive and not really needed taking into account the way how parking works in vanilla

**Comment by Elesbaan70** [comment]
The reason I think this is a TMPE bug is that it doesn't happen without TMPE enabled. Seems pretty straightforward......

**Comment by krzychu124** [comment]
see #1101

**Comment by Elesbaan70** [comment]
I'm confused. Why does this only happen when TMPE is enabled?

**Comment by krzychu124** [comment]
It's because TM:PE is greatly increasing range where vehicles can be parked. That's all

**Comment by krzychu124** [comment]
Yes, I'd say we could skip it when not enabled but I'm not 100% sure if it won't break anything.

**Comment by Bigs2013** [comment]
Im sorry but this is hilarious lmao

**Comment by krzychu124** [comment]
Hehe, yes, but you can fix it forever in one click: "Clear parked cars" at _Maintenance_ tab.
